src/test/token/ERC20Gauges.t.sol

```diff
@@ -0,0 +1,533 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.10;
+
+import {DSTestPlus} from "solmate/test/utils/DSTestPlus.sol";
+import {MockERC20Gauges} from "../mocks/MockERC20Gauges.sol";
+
+contract ERC20GaugesTest is DSTestPlus {
+
+    MockERC20Gauges token;
+    address constant gauge1 = address(0xDEAD);
+    address constant gauge2 = address(0xBEEF);
+
+    function setUp() public {
+        token = new MockERC20Gauges(address(this));
+        token.mint(address(this), 100e18);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        TEST ADMIN GAUGE OPERATIONS
+    //////////////////////////////////////////////////////////////*/
+
+    function testSetMaxGauges() public {
+        token.setMaxGauges(2);
+        require(token.maxGauges() == 2);
+    }
+
+    function testSetMaxUnder() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        hevm.expectRevert(abi.encodeWithSignature("MaxGaugeError()"));
```

Comment:
You can just use `bytes` instead of `abi.encodeWithSignature` I think - I use it

Reply:
> The error data would be encoded identically as the ABI encoding for function calls, i.e., abi.encodeWithSignature("InsufficientBalance(uint256,uint256)", balance[msg.sender], amount).

https://blog.soliditylang.org/2021/04/21/custom-errors/